Common/src/main/java/com/techarium/techarium/inventory/ExchangeStationMenu.java

```diff
@@ -0,0 +1,50 @@
+package com.techarium.techarium.inventory;
+
+import com.techarium.techarium.blockentity.selfdeploying.ExchangeStationBlockEntity;
+import com.techarium.techarium.registry.TechariumMenuTypes;
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.SimpleContainer;
+import net.minecraft.world.entity.player.Inventory;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.inventory.Slot;
+import net.minecraft.world.item.ItemStack;
+
+public class ExchangeStationMenu extends MachineMenu {
+
+	private final SimpleContainer input;
+	private final SimpleContainer output;
+	private final ExchangeStationBlockEntity exchangeStation;
+
+	public ExchangeStationMenu(int id, Inventory playerInventory, Player player, BlockPos pos) {
+		super(TechariumMenuTypes.EXCHANGE_STATION.get(), id, pos);
+		this.exchangeStation = ((ExchangeStationBlockEntity) player.level.getBlockEntity(this.pos));
```

Comment:
This should use the BlockEntityType.getBlockEntity method instead to ensure there will be no cast exception.